### ISSUE TITLE
Discrepancy in records fetched between DB UI and Foxx _query method: Arango v3.7.15

### ISSUE THREAD

**Issue by AnushaVasanth-tech** [arangodb/arangodb#15107]
## My Environment

* __ArangoDB Version__:       3.7.15
* __Storage Engine__:             RocksDB 
* __Deployment Mode__:      Single Server
* __Deployment Strategy__:   ArangoDB Starter (on DEV local) | ArangoDB Starter in Docker (on TEST, UAT, PRD Environment)
* __Configuration__:               
* __Infrastructure__:                own
* __Operating System__:        Windows 10 (on DEV local) | Ubuntu 16.04 (on TEST, UAT, PRD Environment)
* __Total RAM in your machine__:        16 GB (on DEV local) | 62GB n PRD
* __Disks in use__:        
       

## Component, Query & Data

__Affected feature__:
Foxx | AQL query using web interface

__AQL query (if applicable)__:
Attached simplified query 
[simplified_evo_query.txt](https://github.com/arangodb/arangodb/files/7526863/simplified_evo_query.txt)

BindParam for simplified query:
[simplified_evo_bindParam.txt](https://github.com/arangodb/arangodb/files/7526865/simplified_evo_bindParam.txt)

__AQL explain (if applicable)__:
Attached aql execution plan
[simplified_evo_exec_plan.txt](https://github.com/arangodb/arangodb/files/7526877/simplified_evo_exec_plan.txt)

__Dataset__:

__Size of your Dataset on disk__:

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

1. Execute the attached simplified query on Arango web interface, Note the total number of records retrieved
2. Execute the same query from FOXX using db._query method, Capture total number of records retrieved

Following are readings noted in foxx response: 

```
"_countTotal": 6025,
"_countQuery": 6025,
"_skip": 0,
"_limit": null,
"_cached": false,
"_extra": {
    "stats": {
        "writesExecuted": 0,
        "writesIgnored": 0,
        "scannedFull": 9464,
        "scannedIndex": 0,
        "filtered": 275283,
        "httpRequests": 0,
        "executionTime": 1.8366911000339314,
        "peakMemoryUsage": 11007878
    },
    "warnings": []
},
"_current": 0,
"_stop": 6025
}
```
When used "fullCount" option in AQL:
```
"_countTotal": 6025,
"_countQuery": 6025,
"_skip": 0,
"_limit": null,
"_cached": false,
"_extra": {
    "stats": {
        "writesExecuted": 0,
        "writesIgnored": 0,
        "scannedFull": 9464,
        "scannedIndex": 0,
        "filtered": 275283,
        "httpRequests": 0,
        "fullCount": 9464,
        "executionTime": 1.8309500999748707,
        "peakMemoryUsage": 11007878
    },
    "warnings": []
},
"_current": 0,
"_stop": 6025
}

```
__Problem__:

1. Records retrieved by Foxx is always lesser (~80% to 40%) than count available in web interface
2. The result of query is array of objects. When result from FOXX is examined, it was noted that, "null" had been replaced for empty array ( [] ) on random basis for few of nodes.

__Expected result__:
Records retrieved by Foxx should match the total count fetched in Arango web interface (both in count and node content)

__Additional Info__:
Note: The query worked fine on Arango v3.2, 3.3 and 3.6
We have recently upgraded from Arango v3.2.8 (MMFiles) to Arango 3.7.15 (RocksDB). Upgrade was on done on a new VM and data set-up has been redone on new machine.

**Comment by AnushaVasanth-tech** [comment]
Hello,

The result retrieved in foxx matches with UI when disabled the rule "splice-subqueries" in arangod.conf.

As per this, the issues around performance was resolved in 3.7.11 as per "known issues list" (https://www.arangodb.com/docs/stable/release-notes-known-issues37.html ) 
![image](https://user-images.githubusercontent.com/94160854/142875342-78000836-b7d9-44c8-84bf-7b85b0a83079.png)

Could it be that the fix was done only on DB but not on FOXX? It is interesting to note that,  our query had nested sub queries and disabling the rule fixed the issue.

Thanks to confirm the root cause of it, so that it would helpful in further usage of sub-queries

Regards,
Anusha

**Comment by mpoeter** [comment]
@AnushaVasanth-tech apparently there is still some other issue with spliced-subqeries in some edge cases, which your queries triggers. This problem also exists in 3.8 and in current devel. It is still unclear to me though why this is not triggered when runnign the query via the UI. We are investigating this and will keep you posted.

**Comment by AnushaVasanth-tech** [comment]
Hello,

We see that a PR linked to this issue is open since Dec21. Could you please provide by when the fixes will be available in next versions (Arango 3.8+)?

Thank you,

**Comment by mchacki** [comment]
hi sorry,
I think this got automatically closed with the merge of the bug-fix.
So as you see this bug-fix has been merged to 3.8 and will be contained in the next 3.8 release. (and the next 3.9 we are holding back on merges due to current ongoing release, so 3.9.2 should contain the bugfix.)
